src/converse-chatview.js

```diff
@@ -1018,6 +1026,25 @@ converse.plugins.add('converse-chatview', {
                 return this;
             },
 
+            async inject (stanzas) {
+                if (!_converse.debug) {
```

Comment:
Thinking about this a bit more, this feature wouldn't actually have helped you solve the issue with receipts from MAM being processed correctly because you `await` in the `inject` method and that was the problem in the MAM code and this code would have just masked the problem.

Don't be mad at me, but I'm leaning towards rejecting this PR and instead just ask that you write a test in the future, unless you can make a good argument why it's worth having this feature (which needs support going forward and can break when we refactor).

Reply:
I'm absolutely not mad at you. I can follow your reasoning. However I used that feature for tracking down another bug unrelated to the MAM issue. I found that empty messages with just chat-states (see example below) would generate errors. I still think that this feature might be useful for tracking down these kinds of errors:

```xml
<message from="csztest@scholzbande.de" to="csztest@scholzbande.de/converse.js-9009601" xmlns="jabber:client">
    <result id="1566456555786157" queryid="35fcabfb-bbe9-4b9d-957a-83bb1c4df8a8" xmlns="urn:xmpp:mam:2">
        <forwarded xmlns="urn:xmpp:forward:0">
            <message from="csztest@scholzbande.de/converse.js-52677026" id="5b8e6d92-d1ff-4195-afdd-958df0d518c7" to="christoph@scholzbande.de" type="chat" xml:lang="de" xmlns="jabber:client">
                <archived by="csztest@scholzbande.de" id="1566456555786157" xmlns="urn:xmpp:mam:tmp"/>
                <stanza-id by="csztest@scholzbande.de" id="1566456555786157" xmlns="urn:xmpp:sid:0"/>
                <active xmlns="http://jabber.org/protocol/chatstates"/>
                <received id="aa466f2b-2943-4109-9da6-7220285403b6" xmlns="urn:xmpp:receipts"/>
                <store xmlns="urn:xmpp:hints"/>
            </message>
            <delay from="scholzbande.de" stamp="2019-08-22T06:49:15.786157Z" xmlns="urn:xmpp:delay"/>
        </forwarded>
    </result>
</message>
```

Reply:
Ok, so in my opinion, the best way to track down and fix such a bug is to write a test that passes in this message and then to check what happens.

For example, here's a test that checks that a malformed message is rejected:
https://github.com/conversejs/converse.js/blob/bce8dc9113e802e184e31ab52272a5bd3bebf164/spec/muc_messages.js#L14

You could fairly easily create a similar test and just replace the stanza with your own.

This has the added benefit of automatically adding a regression test for any fix you make.

Concerning this PR, I'm going to reject it because it's IMO a very narrow use-case and because it adds more code that needs to be kept up to date and maintained.

You can still put all these changes in a 3rd party plugin. You can use `overrides` to modify `parseMessageForCommands` and the `inject` method can  be assigned with `Object.assign` to `_converse.ChatBox.prototype`. There are various examples of this in the core plugins. Feel free to ask if it's still unclear how to do this.

You can put that plugin in the [community-plugins](https://github.com/conversejs/community-plugins) repo. Maybe eventually other debug functions can be added to it as well.

Thanks for making the PR and for implementing my suggestions.